src/network/netdev/wireguard.c

```diff
@@ -0,0 +1,714 @@
+                          r = sd_netlink_message_open_array(message, ++j);
+                          if (r < 0)
+                                  break;
+
+                          r = sd_netlink_message_append_u16(message, WGALLOWEDIP_A_FAMILY, mask->family);
+                          if (r < 0)
+                                  break;
+
+                          if (mask->family == AF_INET) {
+                                  r = sd_netlink_message_append_in_addr(message, WGALLOWEDIP_A_IPADDR, &mask->ip4);
+                                  if (r < 0)
+                                          break;
+                          } else if (mask->family == AF_INET6) {
+                                  r = sd_netlink_message_append_in6_addr(message, WGALLOWEDIP_A_IPADDR, &mask->ip6);
+                                  if (r < 0)
+                                          break;
+                          }
+
+                          r = sd_netlink_message_append_u8(message, WGALLOWEDIP_A_CIDR_MASK, mask->cidr);
+                          if (r < 0)
+                                  break;
+
+                          r = sd_netlink_message_close_container(message);
+                          if (r < 0)
+                                  return log_netdev_error_errno(netdev, r, "Could add wireguard allowed ip: %m");
+                  }
+                  mask_start = mask;
+                  if (mask_start) {
+                          r = sd_netlink_message_cancel_array(message);
+                          if (r < 0)
+                                  return log_netdev_error_errno(netdev, r, "Could not cancel wireguard allowed ip message attribute: %m");
+                  }
+                  r = sd_netlink_message_close_container(message);
+                  if (r < 0)
+                          return log_netdev_error_errno(netdev, r, "Could not add wireguard allowed ip: %m");
+
+                  r = sd_netlink_message_close_container(message);
+                  if (r < 0)
+                          return log_netdev_error_errno(netdev, r, "Could not add wireguard peer: %m");
+          }
+
+          peer_start = peer;
+          if (peer_start && !mask_start) {
+                  r = sd_netlink_message_cancel_array(message);
+                  if (r < 0)
+                          return log_netdev_error_errno(netdev, r, "Could not cancel wireguard peers: %m");
+          }
+
+          r = sd_netlink_message_close_container(message);
+          if (r < 0)
+                  return log_netdev_error_errno(netdev, r, "Could close wireguard container: %m");
+
+          r = sd_netlink_send(netdev->manager->genl, message, &serial);
+          if (r < 0)
+                  return log_netdev_error_errno(netdev, r, "Could not set wireguard device: %m");
+
+        } while (peer || mask_start);
+
+        return 0;
+}
+
+static WireguardEndpoint* endpoint_free(WireguardEndpoint *e) {
```

Comment:
I'd stick to unified nomenclature and name this wireguard_endpoint_free(), i.e. not drop the prefix here